ui/src/app/views/notebook-redirect/component.spec.ts

```diff
@@ -105,140 +105,289 @@ describe('NotebookRedirectComponent', () => {
+    expect(fixture.debugElement.queryAll(By.css('.notebooks-header'))[0].nativeElement.textContent)
+      .toMatch('Loading Notebook \'foo.ipynb\'');
+  }));
 
-    fixture.componentInstance.notebookName = '1%2B1.ipynb';
-    blockingClusterStub.cluster.status = ClusterStatus.Running;
+  it('should return to the notebooks list page if we cancel notebook creation', fakeAsync(() => {
+    spyOn(window, 'setTimeout').and.stub();
+    updateAndTick(fixture);
+    simulateClick(fixture, fixture.debugElement.queryAll(By.css('#cancelButton'))[0]);
     tick(10000);
+
+    // expect notebook not to have loaded
     fixture.detectChanges();
+    expect(fixture.debugElement.queryAll(By.css('.i-frame')).length).toBe(0);
+  }));
+});
+
+describe('NotebookRedirectComponent', () => {
+  let fixture: ComponentFixture<NotebookRedirectComponent>;
+  let blockingClusterStub: BlockingClusterStub;
+  let blockingNotebooksStub: BlockingNotebooksStub;
+
+  beforeEach(fakeAsync(() => {
+    blockingClusterStub = new BlockingClusterStub();
+    blockingClusterStub.cluster.status = ClusterStatus.Creating;
+    blockingNotebooksStub = new BlockingNotebooksStub();
 
-    expect(fakeWindow.location.href).toContain('/1%252B1.ipynb');
+    TestBed.configureTestingModule({
+      declarations: [
+        NotebookComponent,
+        NotebookRedirectComponent,
+        TopBoxComponent
+      ],
+      imports: [
+        BrowserAnimationsModule,
+        RouterTestingModule,
+        ClarityModule.forRoot()
+      ],
+      providers: [
+        { provide: ClusterService, useFactory: () => blockingClusterStub },
+        { provide: LeoClusterService, useValue: new LeoClusterServiceStub() },
+        { provide: NotebooksService, useFactory: () => blockingNotebooksStub },
+        { provide: JupyterService, useValue: new JupyterServiceStub() },
+        { provide: ActivatedRoute, useValue: {
+          snapshot: {
+            params: {
+              'ns': WorkspaceStubVariables.DEFAULT_WORKSPACE_NS,
+              'wsid': WorkspaceStubVariables.DEFAULT_WORKSPACE_ID,
+              'nbName': '1%2B1'
+            },
+            queryParams: {
+              'kernelType': Kernels.R,
+              'creating': true
+            },
+            queryParamMap: convertToParamMap({
+              'kernelType': Kernels.R,
+              'creating': true
+            })
+          }
+        }},
+      ]}).compileComponents().then(() => {
+      fixture = TestBed.createComponent(NotebookRedirectComponent);
+      spyOn(window.history, 'replaceState').and.stub();
+      blockingClusterStub.release();
+      blockingNotebooksStub.release();
+    });
+  }));
+
+
+  it('should escape notebook names', fakeAsync(() => {
+    blockingClusterStub.cluster.status = ClusterStatus.Running;
+    updateAndTick(fixture);
+    fixture.detectChanges();
+    tick(10000);
+    expect(fixture.debugElement.queryAll(By.css('#leo-iframe'))[0]
+      .properties['src']).toMatch(/1\+1.ipynb/);
```

Comment:
If this assertion is actually passing then I'm perplexed, I would have expected the doubly encoded form. Yet from the manual testing you described it sounds like things are working properly.